### ISSUE TITLE
ModularAHU models fail in OpenModelica

### ISSUE THREAD

**Issue by FWuellhorst** [RWTH-EBC/AixLib#1345]
**Describe the bug**
The models in the package `ModularAHU` fail in OpenModelica.

**Error logs**

```
 Error in model:  AixLib.Systems.ModularAHU.Examples.Cooler 
Simulation execution failed for model: AixLib.Systems.ModularAHU.Examples.Cooler
assert            | warning | The following assertion has been violated during initialization at time 0.000000
|                 | |       | noEvent(registerModule.hydraulicModule.PumpInterface.pump.vol.dynBal.medium.T >= 272.15)
assert            | error   | In AixLib.Systems.ModularAHU.Examples.Cooler.PumpInterface.pump.vol.dynBal.medium: Temperature T = -2.40663e+16 K exceeded its minimum allowed value of -1 degC (272.15 Kelvin) as required from medium model "AixLib.Media.Water".
assert            | info    | simulation terminated by an assertion at initialization

Warning: The model contains alias variables with redundant start and/or conflicting nominal values. It is recommended to resolve the conflicts, because otherwise the system could be hard to solve. To print the conflicting alias sets and the chosen candidates please use -d=aliasConflicts.


 
 Error in model:  AixLib.Systems.ModularAHU.Validation.AHU2_Cooler 
Failed to build model: AixLib.Systems.ModularAHU.Validation.AHU2_Cooler
Error: Too few equations, under-determined system. The model has 2418 equation(s) and 2419 variable(s).


 
 Error in model:  AixLib.Systems.ModularAHU.Validation.AHU2_Heater 
Failed to build model: AixLib.Systems.ModularAHU.Validation.AHU2_Heater
Error: Too few equations, under-determined system. The model has 2416 equation(s) and 2417 variable(s).


 
 Error in model:  AixLib.Systems.ModularAHU.Validation.AHU2_Preheater 
Failed to build model: AixLib.Systems.ModularAHU.Validation.AHU2_Preheater
Error: Too few equations, under-determined system. The model has 2554 equation(s) and 2555 variable(s).

 Error in model:  AixLib.Systems.ModularAHU.GenericAHU 
[D:/04_git/AixLib/AixLib/Fluid/Humidifiers/GenericHumidifier_u.mo:39:60-40:61:writable] Warning: Conditional component ‘Tsteam‘ is used in a non-connect context.
[D:/05_Programme/OpenModelica1.19.2-64bit/lib/omlibrary/Modelica 4.0.0/Blocks/Math.mo:543:5-544:48:writable] Warning: The following equation is INCONSISTENT due to specified unit information: genericAHU.humidifier.gai.k = genericAHU.humidifier.mWat_flow_nominal
Warning: The units of following sub-expressions need to be equal:
- sub-expression "genericAHU.humidifier.mWat_flow_nominal" has unit "kg/s"
- sub-expression "genericAHU.humidifier.gai.k" has unit "1"
[D:/04_git/AixLib/AixLib/Fluid/Humidifiers/GenericHumidifier_u.mo:39:60-40:61:writable] Warning: Conditional component ‘Tsteam‘ is used in a non-connect context.
[D:/05_Programme/OpenModelica1.19.2-64bit/lib/omlibrary/Modelica 4.0.0/Blocks/Math.mo:543:5-544:48:writable] Warning: The following equation is INCONSISTENT due to specified unit information: genericAHU.humidifier.gai.k = genericAHU.humidifier.mWat_flow_nominal
Warning: The units of following sub-expressions need to be equal:
- sub-expression "genericAHU.humidifier.mWat_flow_nominal" has unit "kg/s"
- sub-expression "genericAHU.humidifier.gai.k" has unit "1"
[D:/05_Programme/OpenModelica1.19.2-64bit/lib/omlibrary/Modelica 4.0.0/Blocks/Math.mo:543:5-544:48:writable] Warning: The following equation is INCONSISTENT due to specified unit information: genericAHU.humidifierRet.gai.k = genericAHU.humidifierRet.mWat_flow_nominal
Warning: The units of following sub-expressions need to be equal:
- sub-expression "genericAHU.humidifierRet.mWat_flow_nominal" has unit "kg/s"
- sub-expression "genericAHU.humidifierRet.gai.k" has unit "1"
Error: Internal error Instantiation of AixLib.Systems.ModularAHU.GenericAHU failed with no error message.
```

**Comment by KremerMartin** [comment]
This  might be in parts maybe solved by #1368 and #1346 . I will check that after merging the others.

**Comment by KremerMartin** [comment]
Generic-AHU-example simulates in OM now. Only error message is known from Dymola as well:
```
The following assertion has been violated during initialization at time 0.000000
genericAHU.fanRet.nominalValuesDefineDefaultPressureCurve
*** Warning: You are using a flow or pressure controlled mover with the 
              default pressure curve. 
              This leads to approximate calculations of the electrical power 
              consumption. Add the correct pressure curve in the record per 
              to obtain an accurate computation. 
              Setting nominalValuesDefineDefaultPressureCurve=true will suppress this warning.****
```

**Comment by KremerMartin** [comment]
`AixLib.Systems.ModularAHU.Examples.EONERC_AHU2` simulates in OpenModelica only when using `AixLib.Fluid.FixedResistances.SimplePipe`. But when using `AixLib.Fluid.FixedResistances.PlugFlowPipe` the simulation terminate at time 585 s because of a reinitialization in the delay part. (`plugFlowPipe.del.x`)
I am not sure what exactly is happening there. But since the `AixLib.Fluid.FixedResistances.PlugFlowPipe` is part of the IBPSA, I think we cannot fix it here.

**Comment by KremerMartin** [comment]
The following examples do not simulate in OpenModelica:

- `AixLib.Systems.ModularAHU.Validation.AHU2_Cooler`
- `AixLib.Systems.ModularAHU.Validation.AHU2_Heater`

The error message is as follows (examplary for Heater):
```
Removed more then one event from spatialDistribution. Step size to big!
time: 3628.800000, spatialDistribution index: 11, number of events: 132
Fatal error on line 2608, column 10: tag mismatch
```

I have no clue what might be the problem here (maybe it is caused by the big CombiTimeTable for the measurement results). @FWuellhorst Do you have an idea?

**Comment by FWuellhorst** [comment]
I don't have any idea, but as long as the simulation finishes, I guess it's ok. Dymola throws such errors from time to time as well.
Comparing Dymola to OM for the outputs, I don't see any difference in the outputs of the model. Further comparison of all models will be done anyway. 

Once we have everything running and compared for differences to Dymola, we can start to tackle such messages.

![image](https://user-images.githubusercontent.com/22442514/201601127-19e2d209-07da-484f-9fe6-c80c3edbe35b.png)

**Comment by KremerMartin** [comment]
Thanks for checking, @FWuellhorst . :) 
This is now closed by PR #1379
